### PR TITLE
Handle repeated of the same keys for ref_get; tests for redis and shm

### PR DESCRIPTION
**Pull request by ashule01**
## Change description

On previous implementation, if duplicate keys were sent to reference_get, the splitting of keys into redis and shm ref keys used a dictionary that obliterated knowledge of duplicate keys. This bugfix stores the original indices of the keys as a list so when the data is retrieved, it can be referenced more than once if the keys were duplicated

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

https://elementary.atlassian.net/browse/ERS-2842

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: languages/python/atom/element.py [hunk 1]
```diff
@@ -3470,12 +3478,15 @@ def _reference_get_redis(
...
                 # Deserialize the data
-                return_data_index = int(keys[key])
-                return_data[return_data_index] = (
+                datum = (
                     atom_ser.deserialize(ref, method=serialization)
                     if ref is not None
                     else None
                 )
+                # format return data to same shape as input keys
+                return_data_indices = [int(k) for k in keys[key]]
```
Comment (slaigai) [inline languages/python/atom/element.py:3487]: NIT: If `keys` is a `dict[str, list[int]]`, it looks like we will not need the `int(k)` conversion (or maybe I missed something!)

File: languages/python/atom/element.py [hunk 2]
```diff
@@ -3538,8 +3549,9 @@ def _reference_get_shared_mem(
                             f"An exception occurred when getting data for handle {handle}: {str(e)}"
                         )
 
-                return_data_index = int(keys[key])
-                return_data[return_data_index] = data
+                return_data_indices = [int(k) for k in keys[key]]
```
Comment (slaigai) [inline languages/python/atom/element.py:3552]: NIT: Is the `int(k)` casting necessary here?

File: languages/python/tests/test_atom.py [hunk 3]
```diff
@@ -1707,6 +1707,38 @@ def test_reference_multiple_shared_mem(self, caller):
         assert success == True
         assert len(failed) == 0
 
+    def test_reference_duplicated_keys_redis(self, caller):
```
Comment (slaigai) [inline languages/python/tests/test_atom.py:1710]: NIT: Code can be reused for `test_reference_duplicated_keys` using test parameters. Looks like the only real difference is the `reference_create` call

**Review by slaigai — APPROVED** [review]
Looks solid, only NITs!

File: languages/python/tests/test_atom.py (hunk 3 above)
Comment (ashule01) [reply languages/python/tests/test_atom.py:1710]: Great call on this, updating the test. Thanks Steven!

**Review by aywaldron — APPROVED** [review]
Looks great, thanks!

**Pull request merged**